packages/account/src/BiconomySmartAccount.ts

```diff
@@ -0,0 +1,173 @@
+import { JsonRpcProvider, Web3Provider, Provider } from '@ethersproject/providers'
+import { Signer, ethers, BigNumberish, BytesLike, BigNumber } from 'ethers'
+import { SmartAccount, DEFAULT_USER_OP } from './BaseAccount'
+import { BiconomyPaymasterAPI } from '@biconomy/paymaster'
+import {
+  NODE_CLIENT_URL,
+  EntryPoint_v100__factory,
+  SmartAccountFactory_v100,
+  SmartAccountFactory_v100__factory,
+  SmartAccount_v100__factory,
+} from '@biconomy/common'
+import { BiconomySmartAccountConfig } from './utils/Types'
+import { UserOperation, Transaction } from '@biconomy/core-types'
+import NodeClient from '@biconomy/node-client'
+import INodeClient from '@biconomy/node-client'
+import { IBiconomySmartAccount } from 'interfaces/IBiconomySmartAccount'
+import { SupportedChainsResponse, BalancesResponse, BalancesDto, UsdBalanceResponse, SmartAccountByOwnerDto, SmartAccountsResponse, SCWTransactionResponse } from "@biconomy/node-client"
+
+export class BiconomySmartAccount extends SmartAccount implements IBiconomySmartAccount {
+  private factory: SmartAccountFactory_v100
+  private nodeClient: INodeClient
+
+  constructor(readonly biconomySmartAccountConfig: BiconomySmartAccountConfig) {
+    const { signerOrProvider, rpcUrl, epAddress, factoryAddress, bundlerUrl, paymasterUrl, nodeClientUrl } = biconomySmartAccountConfig
```

Comment:
it could be optional to override?
some reference below

export interface BaseApiParams {
  owner: Signer
  index?: number
  provider: Provider
  entryPointAddress: string
  accountAddress?: string
  overheads?: Partial<GasOverheads>
  paymasterAPI?: PaymasterAPI
  httpRpcClient?: HttpRpcClient
}

export interface ABCAccountApiParams extends BaseApiParams {
  owner: Signer
  index?: number
  factoryAddress?: string
}